virtualizarr/accessor.py

```diff
@@ -48,10 +52,19 @@ def to_icechunk(
 
         If `append_dim` is provided, the virtual dataset will be appended to the existing IcechunkStore along the `append_dim` dimension.
 
+        If `last_updated_at` is provided, it will be used as a checksum for any virtual chunks written to the store with this operation.
+        At read time, if any of the virtual chunks have been updated since this provided datetime, an error will be raised.
+        This protects against reading outdated virtual chunks that have been updated since the last read. When not provided, no check is performed.
+
         Parameters
         ----------
         store: IcechunkStore
         append_dim: str, optional
+            When provided, specifies the dimension along which to append the virtual dataset.
+        last_updated_at: datetime, optional
+            When provided, uses provided datetime as a checksum for any virtual chunks written to the store with this operation.
+            When not provided (default), no check is performed.
+
```

Comment:
Will this work or does it need to be one second in the future?

```suggestion

Examples
--------
To ensure an error is raised if the files containing referenced virtual chunks are modified at any time from now on, pass the current time to ``last_updated_at``.

>>> from datetime import datetime
>>>
>>> vds.virtualize.to_icechunk(
...     icechunkstore,
...     last_updated_at=datetime.now(),
... )
```

Reply:
I guess alternatively we might imagine finding the time at which the files were last modified with `os.stat`, then passing that.

Reply:
Yeah this works in practice, as does using the actual stamp of the file (typically) as long as your round up to the nearest second. Icechunk only store second level precision, so thats why the test adds a second because the tets are so fast it is invalid otherwise.